crypto/crypto.go

```diff
@@ -111,52 +104,27 @@ func CompatibleAlgorithms(sigAlgo SignatureAlgorithm, hashAlgo HashAlgorithm) bo
 
 // A PrivateKey is a cryptographic private key that can be used for in-memory signing.
 type PrivateKey struct {
-	privateKey crypto.PrivateKey
-}
-
-// Sign signs the given message with this private key and the provided hasher.
-//
-// This function returns an error if a signature cannot be generated.
-func (sk PrivateKey) Sign(message []byte, hasher Hasher) ([]byte, error) {
-	return sk.privateKey.Sign(message, hasher)
+	crypto.PrivateKey
 }
 
 // Algorithm returns the signature algorithm for this private key.
 func (sk PrivateKey) Algorithm() SignatureAlgorithm {
-	return SignatureAlgorithm(sk.privateKey.Algorithm())
+	return SignatureAlgorithm(sk.PrivateKey.Algorithm())
 }
```

Comment:
The embedding avoids redefining the same functions, it doesn't work here because of the output type.